using-timescaledb/continuous-aggregates.md

```diff
@@ -96,6 +96,120 @@ with PostgreSQL. This means that aggregates with `ORDER BY` clauses or `DISTINCT
 additionally we require that no `FILTER` clauses appear ([though we will allow them in future
 releases](#future-work)).
 
+### Integer-based Continuous Aggregates [](create-integer)
+
+Usually, continuous aggregates are defined on a time column, but it is
+also possible to create your own custom scheme for handling
+aggregation for tables that are using an integer for the time
+column. This can be useful if you have tables that use other measures
+of time that can be represented as integer values, such as nanosecond
+epochs, minutes since founding date, or whatever is suitable for your
+application.
+
+As an example, suppose that you have a table with CPU and disk usage
+for some devices where time is measured in
+[microfortnights][fff-system]. Since we are using an integer-valued
+column as time, we need to provide the chunk time interval when
+creating the hypertable. In this case, we let each chunk consist of a
+millifortnight.
+
+```sql
+CREATE TABLE devices(
+  time BIGINT,        -- Timestamp in microfortnights
+  cpu_usage INTEGER,  -- Total CPU usage
+  disk_usage INTEGER, -- Total disk usage
+  PRIMARY KEY (time)
+);
+
+SELECT * FROM create_hypertable('devices', 'time',
+                                chunk_time_interval => 1000);
+```
+
+To define a continuous aggregate on a table that is using an integer
+dimension, it is necessary to have a function to get the current
+timestamp in whatever representation that you are using and set it for
+the hypertable using
+[`set_integer_now_func`][api-set-integer-now-func]. The function can
+be defined as a normal Postgres function, but need to be
+[`STABLE`][pg-func-stable], take no arguments, and should return an
+integer of the same type as the time column in the table. In our case,
+this should suffice:
+
+```sql
+CREATE OR REPLACE FUNCTION current_microfortnight() RETURNS BIGINT
+LANGUAGE SQL STABLE AS $$
+	SELECT CAST(1209600 * EXTRACT(EPOCH FROM CURRENT_TIMESTAMP) / 1000000 AS BIGINT)
+$$;
+
+SELECT set_integer_now_func('devices', 'current_microfortnight');
+```
+
+Once the replacement for current timestamp has been set up, you can
+define the continuous aggregate for the `devices` table.
+
+```sql
+CREATE VIEW devices_summary
+WITH (
+  timescaledb.continuous,
+  timescaledb.refresh_interval='1 minutes',
+  timescaledb.refresh_lag = '-5')
+AS SELECT time_bucket('5', time) AS bucket,
```

Comment:
@gayyappan One thing that is not entirely clear here is how `refresh_lag` is used together with the integer-now function, and I think this needs to be added. For inserted rows, are they added when

1. `row.time <= current_fortnight() + refresh_lag`
2. `row.time <= current_fortnight() - refresh_lag`.

The `refresh_lag` is negative here (it was copied from the test), which suggest 1, but usual use of refresh lag suggest 2.

Reply:
It appears to be 1, from `api.md` at least.